enhancements/installer/aws-load-balancer-type.md

```diff
@@ -0,0 +1,288 @@
+
+OR
+
+Option 2:
+```go
+type IngressSpec struct {
+// platformSpec holds desired information specific to the underlying
+// infrastructure provider related to ingress routing.
+PlatformSpec PlatformSpec `json:"platformSpec,omitempty"`
+
+// AWS contains settings specific to the Amazon Web Services infrastructure provider 
+// related to ingress routing.
+// +optional
+AWS *AWSPlatformSpec `json:"aws,omitempty"`
+}
+
+// AWSPlatformSpec holds the desired state of the Amazon Web Services infrastructure provider 
+// related to ingress routing.
+// This only includes fields that can be modified in the cluster.
+type AWSPlatformSpec struct {
+// awsLBType is used to set type of the LB either to CLB/ELB or NLB for the 
+// default ingress controller in AWS environment.
+// +optional
+
+AwsLBType `json:"awsLBType,omitempty"`
+}
+
+```
+
+OR
+
+Option 3:
+If we set it in the infra then the API will look as follows -
+Code will be [here](https://github.com/openshift/api/blob/63b58097a95d1b053cf038f56f89afc42524f598/config/v1/types_infrastructure.go#L351)
+```go
+// AWSPlatformSpec holds the desired state of the Amazon Web Services infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type AWSPlatformSpec struct {
+// awsLBType is used to set type of the LB either to CLB/ELB or NLB for the 
+// default ingress controller in AWS environment.
+// +optional
+
+AwsLBType `json:"awsLBType,omitempty"`	
+}
+```
+
+#### Resource provided to the installer
+
+The users provide a load balancer type to the installer.
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  aws:
+    region: us-west-2
+    awsLBType: nlb
+    subnets:
+    - subnet-1
+    - subnet-2
+    - subnet-3
+pullSecret: '{"auths": ...}'
+sshKey: ssh-ed25519 AAAA...
+```
+
+
+#### Resources created by the installer
+
+- The installer will create ingress cluster object with the load balancer type defined in it.
+  Code changes will be made [here](https://github.com/openshift/installer/blob/2d195fb9a0fc7d291a519772a51cf150b82107ad/pkg/asset/manifests/ingress.go#L111)
+
+```yaml
+oc get ingress.config.openshift.io/cluster -o yaml     
+apiVersion: config.openshift.io/v1
+kind: Ingress
+metadata:
+  creationTimestamp: "2022-04-27T05:28:37Z"
+  generation: 1
+  name: cluster
+  resourceVersion: "26281"
+  uid: bc81f07f-7129-4228-964a-903277f481f5
+spec:
+  awsLBType: nlb
+  domain: apps.miheer411t.0427-cez.qe.rhcloud.com
+```
+
+- The cluster-ingress-operator will check the cluster ingress object for the awsLBType and reconcile to create an default ingress controller
+  with the specified lb type.
+  Code changes will be made [here](https://github.com/openshift/cluster-ingress-operator/blob/9ad500df2a5f0037300f3c3f962132bac4d68263/pkg/operator/controller/ingress/load_balancer_service.go#L381)
+
+
+#### Destroying cluster
+
+- After destroying the cluster the NLB resources created in the AWS cluster will be deleted.
```

Comment:
This isn't a change correct? The NLB resources should already be destroyed unless this is a difference resource. The Machine API creates an NLB for the cluster and those resources should be destroyed.